src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/reducers/__tests__/chartBuilderReducer.test.ts

```diff
@@ -625,14 +614,106 @@ describe('chartBuilderReducer', () => {
     });
   });
 
+  describe('UPDATE_MAP_BOUNDARY_LEVELS', () => {
+    const initialState: ChartBuilderState = {
+      axes: {},
+      options: {
+        height: 300,
+        subtitle: '',
+        title: '',
+        titleType: 'default',
+        alt: '',
+      },
+      map: {
+        dataSetConfigs: [
+          {
+            dataSet: { filters: [] },
+            dataGrouping: { type: 'EqualIntervals', customGroups: [] },
+          },
+        ],
+      },
+    };
+
+    test('sets the boundary level to the payload boundaryLevel', () => {
+      const action: ChartBuilderActions = {
+        type: 'UPDATE_MAP_BOUNDARY_LEVELS',
+        payload: {
+          boundaryLevel: 2,
+          dataSetConfigs: [{ dataSet: { filters: [] }, boundaryLevel: 1 }],
+        },
+      };
+
+      const nextState = produce(
+        chartBuilderReducer({
+          data: [],
+          meta: testFullTable.subjectMeta,
+          tableTitle: undefined,
+        }),
+      )(initialState, action);
+
+      expect(nextState.options?.boundaryLevel).toEqual(2);
+      expect(nextState.map?.dataSetConfigs[0]?.boundaryLevel).toEqual(1);
+    });
+  });
+
+  describe('UPDATE_MAP_DATA_GROUPINGS', () => {
+    const initialState: ChartBuilderState = {
+      axes: {},
+      options: {
+        height: 300,
+        subtitle: '',
+        title: '',
+        titleType: 'default',
+        alt: '',
+      },
+      map: {
+        dataSetConfigs: [
+          {
+            dataSet: { filters: [] },
+            dataGrouping: { type: 'EqualIntervals', customGroups: [] },
+          },
+        ],
+      },
+    };
+    test('sets the dataGrouping to the payload dataGrouping', () => {
+      const testDataGrouping: DataGroupingConfig = {
+        type: 'Custom',
+        customGroups: [{ min: 0, max: 999 }],
+        numberOfGroups: 1,
+      };
+      const action: ChartBuilderActions = {
+        type: 'UPDATE_MAP_DATA_GROUPINGS',
+        payload: {
+          dataSetConfigs: [
+            {
+              dataSet: { filters: [] },
+              dataGrouping: testDataGrouping,
+            },
+          ],
+        },
+      };
+
+      const nextState = produce(
+        chartBuilderReducer({
+          data: [],
+          meta: testFullTable.subjectMeta,
+          tableTitle: undefined,
+        }),
+      )(initialState, action);
+
+      expect(nextState.map?.dataSetConfigs[0]?.dataGrouping).toEqual(
+        testDataGrouping,
+      );
+    });
+  });
```

Comment:
Similar to `UPDATE_MAP_BOUNDARY_LEVELS`, we should have another test case to check that the update doesn't overwrite any other part of the data set (i.e. the boundary level)